src/fsharp/CheckExpressions.fs

```diff
@@ -2592,6 +2592,18 @@ let TcValEarlyGeneralizationConsistencyCheck cenv (env: TcEnv) (v: Val, vrec, ti
     | _ -> ()
 
 
+let obsoleteAllowedToUseObsoleteFilter (env : TcEnv) (op : OperationResult<unit>) : OperationResult<unit> =
+    let filter (warning : exn) =
+        warning.ToString()
+        // here should be traversing the env to see if there's
+        // an enclosing obsolete entity
+        match env.eCallerMemberName with
+        | Some("newAdd") -> false
+        | _ -> true
+    match op with
+    | OkResult(warnings, r) -> OkResult(List.filter filter warnings, r)
+    | ErrorResult(warnings, r) -> ErrorResult(List.filter filter warnings, r)
```

Comment:
`CheckValAttributes` is located in a module above the one with `TcEnv` defined as far as I can see, so I cannot pass an instance of `TcEnv` into it. Instead, I decided to make post-processing. This is a dummy version (it doesn't do anything), but if I understand it correctly, I will have to find out if `env` around the called/addressed symbol is `obsolete` at some point in the hierarchy (any tip is welcomed!)